### PR TITLE
add caching to improve performance

### PR DESCRIPTION
**Pull request by miltontony**

### CODE REVIEW

File: molo/core/models.py [hunk 1]
```diff
@@ -452,7 +453,20 @@ class LanguageRelation(models.Model):
 
 
 class TranslatablePageMixinNotRoutable(object):
+    def get_translation_for_cache_key(self, locale, site, is_live):
+        return "get_translation_for_%s_%s_%s_%s_%s" % (
```
Comment (nathanbegbie) [inline molo/core/models.py]: Could we use `.format` please?

File: molo/core/templatetags/core_tags.py [hunk 2]
```diff
@@ -352,22 +367,19 @@ def get_next_tag(context, tag):
     current_tag = tag.get_main_language_page()
     qs = Tag.objects.descendant_of(request.site.root_page).filter(
         languages__language__is_main_language=True).live()
-    if qs:
+    if qs.exists():
```
Comment (miltontony) [inline molo/core/templatetags/core_tags.py]: This is merely an improvement. Bool on QS is not good.

File: molo/core/templatetags/core_tags.py [hunk 3]
```diff
@@ -352,22 +367,19 @@ def get_next_tag(context, tag):
...
         tags = list(qs)
-        if len(tags) > 1:
```
Comment (miltontony) [inline molo/core/templatetags/core_tags.py]: The `.exists()` makes this if statement redundant. Removed and block below shift indentation to the left.

**Comment by miltontony** [comment]
ready for review

**Comment by saeedmb8** [comment]
👍 🍰 🍰

**Pull request merged**